Lint linter config for Python < 3.6 and issue warning if Actions are using Python 3.6 and older

### Description of the bug

```
Run nf-core -l lint_log.txt lint --dir ${GITHUB_WORKSPACE} --markdown lint_results.md
```
```python
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.6.15/x64/bin/nf-core", line 5, in <module>
    from nf_core.__main__ import run_nf_core
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nf_core/__main__.py", line 19, in <module>
    import nf_core.lint
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nf_core/lint/__init__.py", line 23, in <module>
    import nf_core.modules.lint
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nf_core/modules/__init__.py", line 1, in <module>
    from .bump_versions import ModuleVersionBumper
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nf_core/modules/bump_versions.py", line 24, in <module>
    from .modules_command import ModuleCommand
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nf_core/modules/modules_command.py", line [11](https://github.com/nf-core/scrnaseq/actions/runs/3188310073/jobs/5200822519#step:6:12), in <module>
    from .modules_json import ModulesJson
  File "/opt/hostedtoolcache/Python/3.6.[15](https://github.com/nf-core/scrnaseq/actions/runs/3188310073/jobs/5200822519#step:6:16)/x64/lib/python3.6/site-packages/nf_core/modules/modules_json.py", line 15, in <module>
    import nf_core.modules.modules_repo
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/nf_core/modules/modules_repo.py", line 5, in <module>
    from importlib.resources import path
ModuleNotFoundError: No module named 'importlib.resources'
```
```
Error: Process completed with exit code 1.
```

### Command used and terminal output

```console
https://github.com/nf-core/scrnaseq/actions/runs/3188310073/jobs/5200822519
```


### System information

GHA

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -40,6 +40,7 @@
 - Don't print source file + line number on logging messages (except when verbose) ([#2015](https://github.com/nf-core/tools/pull/2015))
 - Extended the chat notifications to Slack ([#1829](https://github.com/nf-core/tools/pull/1829))
 - Allow other remote URLs not starting with `http` ([#2061](https://github.com/nf-core/tools/pull/2061))
+- Prevent installation with unsupported Python versions ([#2075](https://github.com/nf-core/tools/pull/2075))
 
 ### Modules
 

--- a/setup.py
+++ b/setup.py
@@ -35,7 +35,7 @@ setup(
         "console_scripts": ["nf-core=nf_core.__main__:run_nf_core"],
         "refgenie.hooks.post_update": ["nf-core-refgenie=nf_core.refgenie:update_config"],
     },
-    python_requires='>=3.7, <4',
+    python_requires=">=3.7, <4",
     install_requires=required,
     packages=find_packages(exclude=("docs")),
     include_package_data=True,

--- a/setup.py
+++ b/setup.py
@@ -35,6 +35,7 @@ setup(
         "console_scripts": ["nf-core=nf_core.__main__:run_nf_core"],
         "refgenie.hooks.post_update": ["nf-core-refgenie=nf_core.refgenie:update_config"],
     },
+    python_requires='>=3.7, <4',
     install_requires=required,
     packages=find_packages(exclude=("docs")),
     include_package_data=True,

